osgeo_importer/views.py

```diff
@@ -79,40 +86,50 @@ class FileAddView(FormView, ImportHelper, JSONResponseMixin):
-        outpath = os.path.join('/uploads',str(upload.pk))
-        outdir = os.path.join(settings.MEDIA_ROOT,outpath)
+        outpath = os.path.join('osgeo_importer_uploads',str(upload.pk))
+        outdir = os.path.join(FileSystemStorage().location,outpath)
+        if not os.path.exists(outdir):
+            os.makedirs(outdir)
+        logger.debug("%s",outdir)
+        logger.debug("cleaned_data: %s",form.cleaned_data)
         
         # Move all files to uploads directory using upload pk
         # Must be done for all files before saving upfile for validation
-        for each in form.cleaned_data['files']:
-            shutil.move(each.path, os.path.join(outpath,each.name))
+        finalfiles = []
+        for each in form.cleaned_data['file']:
+            tofile = os.path.join(outdir,os.path.basename(each.name))
+            logger.debug('moving %s to %s', each.name, tofile)
+            shutil.move(each.name, tofile)
+            finalfiles.append(tofile)
 
         # Loop through and create uploadfiles and uploadlayers
-        for each in form.cleaned_data['files']:
-            upfile = UploadFile(upload=upload)
-            upfile.file.name = os.path.join(outpath,each.name)
+        uplayers=[]
+        for each in finalfiles:
+            upfile = UploadFile.objects.create(upload=upload)
+            upfile.file.name = each
             upfile.save()
-
-            upfile_base, upfile_ext = os.path.splitext(each.name)
-            if upfile_ext.lower() not in ['.xml','.sld','.prj','.dbf','shx']:
-                description = self.get_fields(upload_file.file.path)
+            upfile_basename = os.path.basename(each)
+            upfile_root, upfile_ext = os.path.splitext(upfile_basename)
+            if upfile_ext.lower() not in ['.prj','.dbf','.shx']:
+                description = self.get_fields(each)
                 for layer in description:
                     configuration_options = DEFAULT_LAYER_CONFIGURATION.copy()
                     configuration_options.update({'index': layer.get('index')})
-                    upload.uploadlayer_set.add(UploadLayer(name=layer.get('name'),
+                    upload.uploadlayer_set.add(UploadLayer(name=upfile_basename,
                                                         fields=layer.get('fields', {}),
                                                         index=layer.get('index'),
                                                         feature_count=layer.get('feature_count',None),
                                                         configuration_options=configuration_options))
         upload.complete = True
         upload.state = 'UPLOADED'
+        upload.save()
 
         if self.json:
-            return self.render_to_json_response({'state': upload.state, 'id': upload.id})
+            return self.render_to_json_response({'state': upload.state, 'id': upload.id, 'count': UploadFile.objects.filter(upload=upload.id).count()})
```

Comment:
This is purely my ignorance, but I want to understand why 'count' is being added to this response. If it is not being specifically needed by any frontends, I'd rather not include it because making it public API means we can't really remove it, and I'm nervous about what happens if the table is large (table scan hanging the view?)